### PR TITLE
Update WMMA capability command for ISA 10+

### PR DESCRIPTION
**Pull request by ellosel**
When using a wavefront size of 64, we should pass the following instruction to `tryAssembler`:

```
"v_wmma_f32_16x16x16_f16 v[0:3], v[8:15], v[16:23], v[0:3]"
```

when using using a wavefront size of 32 we should use:

```
"v_wmma_f32_16x16x16_f16 v[0:7], v[8:15], v[16:23], v[0:7]"
```

edit: We have opted to only support wave64 WMMA instructions for ISA 10+.

**Testing**

Compiler version: `amdclang++ --version`
```
AMD clang version 19.0.0git (ssh://gerritgit/lightning/ec/llvm-project amd-staging-npi 24284 cb5089623b3268b2628b009afa71fbf00ea5d27a)
```

Operating system images: RHEL8, Ubuntu 20, Ubuntu 22

### CODE REVIEW

File: Tensile/Common.py [hunk 1]
```diff
@@ -2048,7 +2048,10 @@ def GetAsmCaps(isaVersion: IsaVersion, compilerVersion: CompilerVersion) -> Dict
     derivedAsmCaps["HasLshlOr"]             = tryAssembler(isaVersion, "v_lshl_or_b32 v47, v36, 0x2, v34")
     derivedAsmCaps["HasSMulHi"]             = tryAssembler(isaVersion, "s_mul_hi_u32 s47, s36, s34")
 
-    derivedAsmCaps["HasWMMA"]               = tryAssembler(isaVersion, "v_wmma_f32_16x16x16_f16 v[0:7], v[8:15], v[16:23], v[0:7]")
+    useWavefront64 = (isaVersion[0] >= 10)
+    inst = "v_wmma_f32_16x16x16_f16 v[0:3], v[8:15], v[16:23], v[0:3]" if useWavefront64 else "v_wmma_f32_16x16x16_f16 v[0:7], v[8:15], v[16:23], v[0:7]"
+    derivedAsmCaps["HasWMMA"]               = tryAssembler(isaVersion, inst)
+
```
Comment (yoichiyoshida) [inline Tensile/Common.py:2054]: we're either setting this flag or not for the tests, we should pick one or the other, instead of a redundant conditional

**Review by yoichiyoshida — CHANGES_REQUESTED** [review]

**Review by babakpst — APPROVED** [review]

**Comment by babakpst** [comment]
@yoichiyoshida Can you please review this PR again? Thanks.

**Review by yoichiyoshida — COMMENTED** [review]
lgtm

**Comment by babakpst** [comment]
Good, can you please remove the change request, so that I can merge it?

**Comment by nakajee** [comment]
@yoichiyoshida 
I think you need to approve this PR to remove change request.

**Review by yoichiyoshida — APPROVED** [review]

**Pull request merged**